zeroex/orderwatch/order_watcher.go

```diff
@@ -264,20 +267,31 @@ func (w *Watcher) cleanupLoop(ctx context.Context) error {
 }
 
 func (w *Watcher) removedCheckerLoop(ctx context.Context) error {
-	for {
-		start := time.Now()
-		if err := w.permanentlyDeleteStaleRemovedOrders(ctx); err != nil {
-			return err
-		}
+	lastDeleted := time.Now()
```

Comment:
We still should call `permanentlyDeleteStaleRemovedOrders` at the start of `removedCheckerLoop`. It is likely that there are stale orders in the database when we are starting up, so we should remove them right away instead of waiting.